recipe/meta.yaml

```diff
@@ -37,7 +37,7 @@ outputs:
     script: build-mpi.sh
     build:
       run_exports:
-        - {{ pin_subpackage('mpich', max_pin='x') }}
+        - mpich >={{ version }},<5.0a0
```

Comment:
MPICH has a commitment to ABI compatibility. v4.0 is supposed to be ABI compatible with any other v4.x build.